src/adc.rs

```diff
@@ -821,6 +914,126 @@ macro_rules! adc_hal {
+                        Sequence::Sixteen  => self.rb.sqr4.modify(|_, w| unsafe {w.sq16().bits(chan) }),
+                    }
+
+                    //Set the sample time for the channel
+                    let st = u8::from(sample_time);
+                    match chan {
+                        0 => self.rb.smpr1.modify(|_, w| w.smp0().bits(st) ),
+                        1 => self.rb.smpr1.modify(|_, w| w.smp1().bits(st) ),
+                        2 => self.rb.smpr1.modify(|_, w| w.smp2().bits(st) ),
+                        3 => self.rb.smpr1.modify(|_, w| w.smp3().bits(st) ),
+                        4 => self.rb.smpr1.modify(|_, w| w.smp4().bits(st) ),
+                        5 => self.rb.smpr1.modify(|_, w| w.smp5().bits(st) ),
+                        6 => self.rb.smpr1.modify(|_, w| w.smp6().bits(st) ),
+                        7 => self.rb.smpr1.modify(|_, w| w.smp7().bits(st) ),
+                        8 => self.rb.smpr1.modify(|_, w| w.smp8().bits(st) ),
+                        9 => self.rb.smpr1.modify(|_, w| w.smp9().bits(st) ),
+                        10 => self.rb.smpr2.modify(|_, w| w.smp10().bits(st) ),
+                        11 => self.rb.smpr2.modify(|_, w| w.smp11().bits(st) ),
+                        12 => self.rb.smpr2.modify(|_, w| w.smp12().bits(st) ),
+                        13 => self.rb.smpr2.modify(|_, w| w.smp13().bits(st) ),
+                        14 => self.rb.smpr2.modify(|_, w| w.smp14().bits(st) ),
+                        15 => self.rb.smpr2.modify(|_, w| w.smp15().bits(st) ),
+                        16 => self.rb.smpr2.modify(|_, w| w.smp16().bits(st) ),
+                        17 => self.rb.smpr2.modify(|_, w| w.smp17().bits(st) ),
+                        18 => self.rb.smpr2.modify(|_, w| w.smp18().bits(st) ),
+                        _ => unimplemented!(),
+                    }
+
+                    // Select channel (with preselection, refer to RM0433 Rev 7 - Chapter 25.4.12)
+                    self.rb.pcsel.modify(|r, w| unsafe { w.pcsel().bits(r.pcsel().bits() | (1 << chan)) });
+
+                }
+
+                /// Starts sequence conversion in DMA mode.
+                ///
+                /// This method starts a conversion sequence with DMA
+                /// enabled. The DMA mode selected depends on the [`AdcDmaMode`] specified.
+                /// Waits for the hardware to indicate it's actually started.
+                #[inline(always)]
+                pub fn start_conversion_sequence_dma(&mut self, mode: AdcDmaMode) {
+                    self.check_conversion_conditions();
+
+                    // Set DMA mode
+                    self.rb.cfgr.modify(|_, w| w.dmngt().bits(match mode {
+                        AdcDmaMode::OneShot => 0b01,
+                        AdcDmaMode::Circular => 0b11,
+                    }));
+
+                    // Set resolution
+                    self.rb.cfgr.modify(|_, w| unsafe { w.res().bits(self.get_resolution().into()) });
```

Comment:
Needs to be unsafe?

Reply:
Yes, a bit weird, but that register was set to unsafe for some reason.